data/datasets/TSSB-3M/generate_dataset.py

```diff
@@ -0,0 +1,133 @@
+"""Convert the source TSSB-3M  dataset to instruction data
+"""
+
+import json
+import re
+from os.path import join
+
+from tqdm import tqdm
+
+TEMPLATE = """User: Find the bug in the following code:
+{}
+Reply: The fixed code is:
+```
+{}
+```
+"""
+
+
+# template for pretty output(multiple lines with `User:` & `Reply`)
+TEMPLATE_COMMIT_MSG = """User: Find the bug in the following code:
+{}
+Reply: The bugfix can be described as follows:
+{}
+The fixed code is:
+```
+{}
+```
+"""
+
+INSTRUCTON_TEMPLATE = """Find the bug in the following code:
+{}
+"""
+
+
+# template for json output(value)
+
+RESPONSE_TEMPLATE = """The fixed code is:
+```
+{}
+```
+"""
+
+RESPONSE_TEMPLATE_COMMIT_MSG = """The bugfix can be described as follows:
+{}
+
+The fixed code is:
+```
+{}
+```
+"""
+
+
+def remove_starting_plus_minus(text):
+    if text.startswith("+") or text.startswith("-"):
+        return text[1:]
+    else:
+        return text
+
+
+def remove_extraneous_diff_info(text):
+    pattern = "@@.*@@"
+    return re.sub(pattern, "", text)
+
+
+def clean(text):
+    return remove_extraneous_diff_info(remove_starting_plus_minus(text))
+
+
+def clean_commit_msg(text):
+    """
+    # remove issue id , eg. msg: "rename (hetr_passes -> passes) #1195" -> "rename (hetr_passes -> passes)"
+    # remove `fix` prefix:
+    ##  eg. [fix] 拼写错误 -> 拼写错误
+    ## eg. [FIX] purchase_indonesia : AttributeError 'NoneType' object has no attribute 'id' ->  AttributeError 'NoneType' object has no attribute 'id'
+    """
+    # Remove issue id
+    # pattern = "#\d{1,6}|[fix]|[FIX]"
+    pattern = r"#\d{1,6}"
+    text = re.sub(pattern, "", text)
```

Comment:
is this safe? wouldn't something like 
`re.sub(r"(.+?\s\(.+?\))\s#\d{1,6}", '\\1', text)`
work better?